sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala

```diff
@@ -66,4 +68,41 @@ class PruneFileSourcePartitionsSuite extends QueryTest with SQLTestUtils with Te
       }
     }
   }
+
+  test("SPARK-20986 Reset table's statistics after PruneFileSourcePartitions rule") {
+    withTempView("tempTbl") {
+      withTable("partTbl") {
+        spark.range(10).selectExpr("id").createOrReplaceTempView("tempTbl")
+        sql("CREATE TABLE partTbl (id INT) PARTITIONED BY (part INT) STORED AS parquet")
+        for (part <- Seq(1, 2)) {
+          sql(
+            s"""
+               |INSERT OVERWRITE TABLE partTbl PARTITION (part='$part')
+               |select id from tempTbl
+            """.stripMargin)
+        }
+
+        val tableName = "partTbl"
+        sql(s"ANALYZE TABLE partTbl COMPUTE STATISTICS")
+        val tableStats =
+          spark.sessionState.catalog.getTableMetadata(TableIdentifier(tableName)).stats
+        assert(tableStats.isDefined && tableStats.get.sizeInBytes > 0, "tableStats is lost")
+
+        withSQLConf(SQLConf.ENABLE_FALL_BACK_TO_HDFS_FOR_STATS.key -> "true") {
+          val df = sql("SELECT * FROM partTbl where part = 1")
+          val query = df.queryExecution.analyzed.analyze
+          val sizes1 = query.collect {
+            case relation: LogicalRelation => relation.computeStats(conf).sizeInBytes
+          }
+          assert(sizes1.size === 1, s"Size wrong for:\n ${df.queryExecution}")
+          assert(sizes1(0) == tableStats.get.sizeInBytes)
+          val sizes2 = Optimize.execute(query).collect {
+            case relation: LogicalRelation => relation.catalogTable.get.stats.get.sizeInBytes
```

Comment:
fixed the wrong place? For `size1`, could you get the catalog stats? We'd better not to `computeStats` for analyzed plan. For `size2` or `size3`, we can get sizes from both the catalog stats and `computeStats`, see if they are equal and larger than `size1`.